### PR TITLE
Fix issues with filename argument to configure_traits

### LINKED ISSUE #571: configure_traits(filename=...) does not work as documented
`configure_traits` is supposed to have the ability to load and store objects from pickles, by specifying the `filename` argument.

There are at least two issues here:

- the current code has an undefined name `sm`, almost certainly accidentally introduced during the `2to3` to `six` transition
- the documentation states that:

>         If the file referenced by *filename* does not exist, the object is not
>         modified before displaying the dialog box.

However, in practice if the file doesn't already exist an error is raised.

### PR DESCRIPTION
**Pull request by mdickinson**
Two fixes to `configure_traits` when used with the `filename=` argument.

- Fixes use of undefined name "sm".
- Allows the specific filename to refer to a file that does not exist, as described in the documentation.
- Adds minimal unit tests, with `view_application` mocked.

Fixes #571

### CODE REVIEW

File: traits/has_traits.py [hunk 1]
```diff
@@ -28,8 +28,9 @@
 from __future__ import absolute_import, division, print_function
 
 import copy as copy_module
-import weakref
+import os
 import re
+import weakref
```
Comment (mdickinson) [inline traits/has_traits.py]: unrelated drive-by alphabetization

File: traits/has_traits.py [hunk 2]
```diff
@@ -2333,8 +2334,9 @@ def configure_traits(
         modified by the user.
         """
         if filename is not None:
-            with open(filename, "rb") as fd:
-                self.copy_traits(sm.cPickle.Unpickler(fd).load())
+            if os.path.exists(filename):
+                with open(filename, "rb") as fd:
+                    self.copy_traits(six.moves.cPickle.Unpickler(fd).load())
```
Comment (rahulporuri) [inline traits/has_traits.py]: do we want to restore the original behavior where we have a try/except around `copy_traits`?

File: traits/tests/test_configure_traits.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+class Model(HasTraits):
+    count = Int
+
+
+@requires_traitsui
+class TestConfigureTraits(unittest.TestCase):
+    def setUp(self):
+        import traitsui.api
```
Comment (rahulporuri) [inline traits/tests/test_configure_traits.py]: any specific reason why you want to import inside the `setUp` instead of just updating the `...optional_dependencies` import?

Comment (mdickinson) [reply traits/tests/test_configure_traits.py]: The `optional_dependencies` only has `traitsui` available, so you'd then have to test whether `traitsui` is `None` and if not, do another import. (`traitsui.api` is not immediately available from an `import traitsui`) It's less complicated this way.

File: traits/has_traits.py (hunk 2 above)
Comment (mdickinson) [reply traits/has_traits.py]: Good point. But no, an exception being raised at that point is likely a coding mistake, so let's allow it to propagate.

**Comment by mdickinson** [comment]
Failures on Python 3.5: `assert_called_once` is new in Python 3.6. Will fix.

File: traits/has_traits.py (hunk 2 above)
Comment (rahulporuri) [reply traits/has_traits.py]: Is it worth it to test/check that the exception is raised?

Comment (mdickinson) [reply traits/has_traits.py]: Perhaps. But I'm not sure I see the value in such a test, except as a regression test, and even then I wouldn't expect new code to go back to suppressing all exceptions.

Comment (mdickinson) [reply traits/has_traits.py]: I'll add a test with a file whose contents aren't a valid pickle. That'll cover the behaviour change from ancient versions where the resulting exception would have been swallowed.
